sympy/functions/special/gamma_functions.py

```diff
@@ -307,6 +307,8 @@ def eval(cls, a, x):
             elif a.is_Integer or (2*a).is_Integer:
                 b = a - 1
                 if b.is_positive:
+                    if a.is_integer:
+                        return factorial(b) - exp(-x) * factorial(b) * sum([x ** k / factorial(k) for k in range(a)])
                     return b*cls(b, x) - x**b * exp(-x)
```

Comment:
The half-integer case remains recursive and will cause the same issues. Why not get rid of it too? With [this formula](http://functions.wolfram.com/GammaBetaErf/Gamma3/16/01/01/0003/), where z1 =0, it reduces to lowergamma(1/2, x) without recursion. 

In face, the TODO on line 302 says this should be done.